src/libcore/num/mod.rs

```diff
@@ -1985,44 +1985,142 @@ $EndFeature, "
             pub fn is_negative(self) -> bool { self < 0 }
         }
 
-        /// Return the memory representation of this integer as a byte array.
+        /// Return the memory representation of this integer as a byte array in
+        /// big-endian (network) byte order.
         ///
-        /// The target platform’s native endianness is used.
-        /// Portable code likely wants to use this after [`to_be`] or [`to_le`].
+        /// # Examples
         ///
-        /// [`to_be`]: #method.to_be
-        /// [`to_le`]: #method.to_le
+        /// ```
+        /// #![feature(int_to_from_bytes2)]
+        ///
+        /// let bytes = 0x12345678i32.to_be_bytes();
+        /// assert_eq!(bytes, [0x12, 0x34, 0x56, 0x78]);
+        /// ```
+        #[unstable(feature = "int_to_from_bytes2", issue = "0")]
+        #[inline]
+        pub fn to_be_bytes(self) -> [u8; mem::size_of::<Self>()] {
+            self.to_be().to_native_bytes()
+        }
+
+        /// Return the memory representation of this integer as a byte array in
+        /// little-endian byte order.
         ///
         /// # Examples
         ///
         /// ```
-        /// let bytes = i32::min_value().to_be().to_bytes();
+        /// #![feature(int_to_from_bytes2)]
+        ///
+        /// let bytes = 0x12345678i32.to_le_bytes();
+        /// assert_eq!(bytes, [0x78, 0x56, 0x34, 0x12]);
+        /// ```
+        #[unstable(feature = "int_to_from_bytes2", issue = "0")]
+        #[inline]
+        pub fn to_le_bytes(self) -> [u8; mem::size_of::<Self>()] {
+            self.to_le().to_native_bytes()
+        }
+
+        /// Return the memory representation of this integer as a byte array in
+        /// native byte order.
+        ///
+        /// As the target platform's native endianness is used, portable code
+        /// should use [`to_be_bytes`] or [`to_le_bytes`], as appropriate,
+        /// instead.
+        ///
+        /// [`to_be_bytes`]: #method.to_be_bytes
+        /// [`to_le_bytes`]: #method.to_le_bytes
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_to_from_bytes2)]
+        ///
+        /// let bytes = i32::min_value().to_be().to_native_bytes();
         /// assert_eq!(bytes, [0x80, 0, 0, 0]);
         /// ```
+        #[unstable(feature = "int_to_from_bytes2", issue = "0")]
+        #[inline]
+        pub fn to_native_bytes(self) -> [u8; mem::size_of::<Self>()] {
+            unsafe { mem::transmute(self) }
+        }
+
+        /// Does the same thing as [`to_native_bytes`].
+        ///
+        /// [`to_native_bytes`]: #method.to_native_bytes
+        #[rustc_deprecated(since = "1.29.0", reason = "method doesn't communicate
+            intent, use `to_native_bytes`, `to_le_bytes` or `to_be_bytes` instead")]
         #[stable(feature = "int_to_from_bytes", since = "1.29.0")]
```

Comment:
If this PR is accepted, we should better keep `to_bytes()`/`from_bytes()` unstable or just remove them, instead of stabilize and immediately deprecate them.